### PR TITLE
validate string lengths where needed

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: src/wire/mod.rs [hunk 1]
```diff
@@ -431,4 +450,31 @@ mod tests {
...
+
+    #[test]
+    fn deserialize_string_within_range_inclusive() {
+        use serde::Deserialize;
+
+        #[derive(Debug, Deserialize, PartialEq, Eq)]
+        struct Test(
+            #[serde(deserialize_with = "super::string_within_range_inclusive::<1, 128, _>")] String,
+        );
+
```
Comment (tdittr) [inline src/wire/mod.rs:462]: Nit: This could also test the too short case

**Review by tdittr — APPROVED** [review]
Looks great, just one nit on the test. (It's already indirectly covered, but testing it should be quick)

File: src/wire/mod.rs (hunk 1 above)
Comment (folkertdev) [reply src/wire/mod.rs:462]: it is? on line 475 we give it a string of length zero

Comment (tdittr) [reply src/wire/mod.rs:462]: oh sorry... I overlooked that completely

**Pull request merged**